To: vim_dev@googlegroups.com
Subject: Patch 7.3.197
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.197
Problem:    When a QuickfixCmdPost event removes all errors, Vim still tries
	    to jump to the first error, resulting in E42.
Solution:   Get the number of error after the autocmd event. (Mike Lundy)
Files:	    src/quickfix.c


*** ../mercurial/vim73/src/quickfix.c	2011-05-10 16:41:13.000000000 +0200
--- src/quickfix.c	2011-05-19 17:34:11.000000000 +0200
***************
*** 2813,2829 ****
  					   (eap->cmdidx != CMD_grepadd
  					    && eap->cmdidx != CMD_lgrepadd),
  					   *eap->cmdlinep);
  #ifdef FEAT_AUTOCMD
      if (au_name != NULL)
  	apply_autocmds(EVENT_QUICKFIXCMDPOST, au_name,
  					       curbuf->b_fname, TRUE, curbuf);
  #endif
      if (res > 0 && !eap->forceit)
-     {
- 	if (wp != NULL)
- 	    qi = GET_LOC_LIST(wp);
  	qf_jump(qi, 0, 0, FALSE);		/* display first error */
-     }
  
      mch_remove(fname);
      vim_free(fname);
--- 2813,2833 ----
  					   (eap->cmdidx != CMD_grepadd
  					    && eap->cmdidx != CMD_lgrepadd),
  					   *eap->cmdlinep);
+     if (wp != NULL)
+ 	qi = GET_LOC_LIST(wp);
  #ifdef FEAT_AUTOCMD
      if (au_name != NULL)
+     {
  	apply_autocmds(EVENT_QUICKFIXCMDPOST, au_name,
  					       curbuf->b_fname, TRUE, curbuf);
+ 	if (qi->qf_curlist < qi->qf_listcount)
+ 	    res = qi->qf_lists[qi->qf_curlist].qf_count;
+ 	else
+ 	    res = 0;
+     }
  #endif
      if (res > 0 && !eap->forceit)
  	qf_jump(qi, 0, 0, FALSE);		/* display first error */
  
      mch_remove(fname);
      vim_free(fname);
*** ../vim-7.3.196/src/version.c	2011-05-19 17:25:36.000000000 +0200
--- src/version.c	2011-05-19 17:41:50.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     197,
  /**/

-- 
I AM THANKFUL...
...for the clothes that fit a little too snug because it
means I have more than enough to eat.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
